c7n/filters/vpc.py

```diff
@@ -29,6 +30,24 @@ class SecurityGroupFilter(RelatedResourceFilter):
     AnnotationKey = "matched-security-groups"
 
 
+class VpcFilter(RelatedResourceFilter):
```

Comment:
this implementation should probably go directly into vpc.py, unlike the other net filters it not generic, its hardcoded to the relationship between vpc/sg